CMakeLists.txt

```diff
@@ -212,13 +210,53 @@ include_directories( ${DBUS_CXX_INCLUDE_DIRECTORIES}
     ${sigc_INCLUDE_DIRS} )
 
 
+#
+# Dependencies
+#
+set( THREADS_PREFER_PTHREAD_FLAG ON )
+find_package( Threads REQUIRED )
+set( DBUS_CXX_LIBRARIES ${sigc_LIBRARIES} Threads::Threads )
+
+find_library( LIBRT rt )
+if( LIBRT )
+	list( APPEND DBUS_CXX_LIBRARIES ${LIBRT} )
+endif( LIBRT )
+
+if( WIN32 )
+    if( CMAKE_SYSTEM_VERSION )
+        set( ver ${CMAKE_SYSTEM_VERSION} )
+        string( REGEX MATCH "^([0-9]+).([0-9])" ver ${ver} )
+        string( REGEX MATCH "^([0-9]+)" verMajor ${ver} )
+        # Check for Windows 10, b/c we'll need to convert to hex 'A'.
+        if( "${verMajor}" MATCHES "10" )
+            set( verMajor "A" )
+            string( REGEX REPLACE "^([0-9]+)" ${verMajor} ver ${ver} )
+        endif()
+        # Remove all remaining '.' characters.
+        string( REPLACE "." "" ver ${ver} )
+        # Prepend each digit with a zero.
+        string( REGEX REPLACE "([0-9A-Z])" "0\\1" ver ${ver} )
+        set( WIN32_VER "0x${ver}" )
+    endif( CMAKE_SYSTEM_VERSION )
+    add_definitions( -D_WIN32_WINNT=${WIN32_VER} )
+    set( CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON )
+
+    if( MSVC )
+        set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /Zc:__cplusplus /std:c++17" )
```

Comment:
Shouldn't this be handled by `set_property( TARGET dbus-cxx PROPERTY CXX_STANDARD 17 )`?  Or does that not work correctly for MSVC?

This is also probably something that should be documented for users, since the main dbus-cxx header does check for this macro, and will fail if C++17 is not available.

Reply:
Will double-check.

Reply:
Removed `/std:c++17` but `/Zc:__cplusplus` must stay unfortunately.